src/masternode/activemasternode.cpp

```diff
@@ -177,8 +177,13 @@ void CActiveMasternodeManager::UpdatedBlockTip(const CBlockIndex* pindexNew, con
 
 bool CActiveMasternodeManager::GetLocalAddress(CService& addrRet)
 {
-    // First try to find whatever local address is specified by externalip option
-    bool fFoundLocal = GetLocal(addrRet) && IsValidNetAddr(addrRet);
+    // First try to find whatever our own local address is known internally.
+    // Addresses could be specified via externalip or bind option, discovered via UPnP
+    // or added by TorController. Use some random dummy IPv4 peer to prefer the one
+    // reachable via IPv4.
+    CNetAddr addrDummyPeer;
+    LookupHost("8.8.8.8", addrDummyPeer, false);
```

Comment:
Can LookupHost fail here? Maybe check and return on failure to be sure

Reply:
Technically, it shouldn't cause otherwise I'd expect `netbase_tests` to fail too. But it's better to be on the safe side, I agree :) pls see f96830b